hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/HoodieSyncConfig.java

```diff
@@ -56,22 +59,14 @@ public class HoodieSyncConfig extends HoodieConfig {
   public static final ConfigProperty<String> META_SYNC_DATABASE_NAME = ConfigProperty
       .key("hoodie.datasource.hive_sync.database")
       .defaultValue("default")
+      .withInferFunction(cfg -> Option.ofNullable(cfg.getString(DATABASE_NAME)))
```

Comment:
We should note that `ConfigProperty#getInferFunc` gets called only in `ConfigProperty#setDefaultValue(ConfigProperty)` so we should check how the default is being set at the call site for all such configs with infer function.

Reply:
good catch. actually these infer functions were never effective as `ConfigProperty#setDefaultValue(ConfigProperty)` wasn't invoked at all. Also noticed that this can be a bug where `org.apache.hudi.common.config.HoodieConfig#setDefaults` never infer defaults if no default value was set. I can file a separate fix for that.